### PR TITLE
Resubscribing Fixes

### PR DESCRIPTION
**Pull request by stephencelis**
We were making a few small bad assumptions here that prevented folks from re-subscribing after cancellation!

### CODE REVIEW

File: Sources/PointFree/Account/Account.swift [hunk 1]
```diff
@@ -451,7 +451,7 @@ private let subscriptionPlanRows = View<Stripe.Subscription> { subscription in
 }
 
 private func mainAction(for subscription: Stripe.Subscription) -> Node {
-  if subscription.cancelAtPeriodEnd {
```
Comment (stephencelis) [inline Sources/PointFree/Account/Account.swift]: This value remains `true` even after cancellation.

File: Sources/PointFree/Pricing.swift [hunk 2]
```diff
@@ -584,8 +584,15 @@ func redirectActiveSubscribers<A>(
           )
           ?? throwE(unit)
 
-        return (userSubscription.run.parallel <|> ownerSubscription.run.parallel)
-          .sequential
+        let race = (userSubscription.run.parallel <|> ownerSubscription.run.parallel).sequential
+
+        return EitherIO(run: race)
+          .flatMap {
+            $0.stripeSubscriptionStatus == .active
```
Comment (stephencelis) [inline Sources/PointFree/Pricing.swift]: @mbrandonw Maybe I am misreading, but looks like the pricing page redirect logic was merely checking for a subscription at all, not whether or not it was active.

File: Sources/PointFreeTestSupport/PointFreeTestSupport.swift [hunk 3]
```diff
@@ -366,7 +369,7 @@ extension Stripe.Subscription {
   public static let canceling = mock
     |> \.cancelAtPeriodEnd .~ true
 
-  public static let canceled = mock
+  public static let canceled = canceling
```
Comment (stephencelis) [inline Sources/PointFreeTestSupport/PointFreeTestSupport.swift]: This immediately made the account page render test for canceling subscriptions fail 😄

File: Tests/PointFreeTests/PricingTests.swift [hunk 4]
```diff
@@ -76,4 +76,17 @@ class PricingTests: TestCase {
 
     assertSnapshot(matching: result.perform())
   }
+
+  func testPricingLoggedIn_CanceledSubscriber() {
```
Comment (stephencelis) [inline Tests/PointFreeTests/PricingTests.swift]: Quick test for pricing redirect logic.

**Comment by stephencelis** [comment]
I've deployed this branch, but will merge before monday if you don't have time to check it out.

File: Sources/PointFree/Pricing.swift (hunk 2 above)
Comment (mbrandonw) [reply Sources/PointFree/Pricing.swift]: On the road so hard to know for sure, but I believe I did that thinking the user would fix their subscription from the account page. 

Guess I’m a lil murky on when a user should go through the pricing page again

Comment (stephencelis) [reply Sources/PointFree/Pricing.swift]: Gotcha. OK I think there's more nuance here to handle `pastDue` accounts! Thanks!

**Comment by stephencelis** [comment]
Gonna merge now to make it easier to work other branches off these fixes.

**Pull request merged**